types.d.ts

```diff
@@ -0,0 +1 @@
+declare type packageManager = 'pnpm' | 'yarn (berry)' | 'yarn (classic)' | 'npm';
```

Comment:
sorry for being nitpicky but could you have this in the `src/utils/getSpawnCommand.ts` file and define it as:
```ts
type PackageManager = keyof typeof packageManagers;
```
?

I think that that would be better because you don't need to repeat anything more than once so it's more robust and maintainable (for example, if in the future we wanted to rename `"yarn (classic)"` to `"yarn 1"`, with this implementation we'd have to change it both here and in the packageManagers object, it's not a big deal since we'd get type errors if we'd forget one but still you get what I mean)

You're also not importing that type anywhere so it's not really going to add extra exports/imports

(also types should be capitalized 😜)

(and finally I am not sure, I think it is quite uncommon to create `.d.ts` for local types? we do that for the vercel types mainly I think because they are shared between the node code and the edge template code, but I don't think we should have other `.d.ts` files if we can help it)